servers/rendering/shader_language.cpp

```diff
@@ -4076,13 +4076,17 @@ PropertyInfo ShaderLanguage::uniform_to_property_info(const ShaderNode::Uniform
 		case ShaderLanguage::TYPE_BOOL:
 			if (p_uniform.array_size > 0) {
 				pi.type = Variant::PACKED_INT32_ARRAY;
+				pi.hint = PROPERTY_HINT_TYPE_STRING;
+				pi.hint_string = itos(Variant::INT) + "/" + itos(PROPERTY_HINT_FLAGS) + ":" + RTR("On");
```

Comment:
Slightly hacky I'd say, but translation works fine with this, so for now it's sufficient, tried with `hint_string = itos(Variant::BOOL)` but it didn't work, unsure if that's because I'm not using it correctly or because it's bugged somehow, but solved it this way for now

Reply:
Could you explain the logic here a bit? Using type string doesn't seem obvious to me as I am a bit unfamiliar with this area of the editor.

Reply:
This makes it a check, a single "on" checkbox as flags, will upload a screenshot when I'm back at the computer

Reply:
![image](https://github.com/godotengine/godot/assets/96648715/2c2ce97b-7528-44f6-a65c-99ecd8047f3a)

Like so, for boolean vectors and arrays of booleans

Reply:
Thank you. I already understand _what_ it does, what I don't understand is why setting the hint to a string and then using that specific string ("2/6:On") results in a checkbox

Reply:
If it helps, the format is `{element_type}/{element_hint}:{element_hint_string}`. This means that for each element in the array, we can specify a `hint` and `hint_string`.

In this case, to export a property as a checkbox you need to use `PROPERTY_HINT_FLAGS` as the `hint`, and then you can specify the label in the `hint_string`.

There's more examples in the [`PROPERTY_HINT_TYPE_STRING`](https://docs.godotengine.org/en/stable/classes/class_@globalscope.html#class-globalscope-constant-property-hint-type-string) documentation.

Reply:
Thank you! That makes sense